### PR TITLE
Update flaky test to only generate postcodes of at least 5 digits

### PR DESCRIPTION
**Pull request by C-Thomp**

### CODE REVIEW

**Review by koladeadewuyi — APPROVED** [review]

File: test/pages/IndDoYouHaveNINumberPageSpec.scala [hunk 1]
```diff
@@ -33,6 +36,17 @@ class IndDoYouHaveNINumberPageSpec extends PageBehaviours {
     booleanField  <- arbitrary[Boolean]
   } yield (addressLookup, address, postcode, name, booleanField)
 
+  private val addressResponseGen: Gen[AddressResponse] = for {
```
Comment (chriswaugh) [inline test/pages/IndDoYouHaveNINumberPageSpec.scala:39]: An AddressResponse generator already exists in test-utils/ModelGenerators - line 149 (it's imported as RegistrationAddressResponse which may have made it evade Cmd+F)

It uses the Address generator at its code so could you update that instead please then it should fix all address for all pages.

**Review by chriswaugh — CHANGES_REQUESTED** [review]

File: test/pages/IndDoYouHaveNINumberPageSpec.scala [hunk 2]
```diff
@@ -33,6 +36,17 @@ class IndDoYouHaveNINumberPageSpec extends PageBehaviours {
...
+    addressline  <- arbitrary[String]
+    addressline2 <- arbitrary[Option[String]]
+    addressline3 <- arbitrary[Option[String]]
+    addressline4 <- arbitrary[Option[String]]
+    postcode     <- Gen.option(alphaNumStr.suchThat(_.length >= 5))
```
Comment (chriswaugh) [inline test/pages/IndDoYouHaveNINumberPageSpec.scala:44]: FWIW - in this case `suchThat` should be fine, but I'd generally favour using a sized generator over this approach. That way you generate something you know will be big/small enough each time. Using `suchThat` means something random will be generated, then if it doesn't match the condition it's thrown away and another random value is generated. Sometimes that can be an issue but in this case it should be fine.

Comment (C-Thomp) [reply test/pages/IndDoYouHaveNINumberPageSpec.scala]: Good point, in the refactor for the about point ive found a nicer way to generate the postcode parameter, tell me what you think :)

**Review by chriswaugh — APPROVED** [review]
*chef's kiss* love it

**Pull request merged**